client/src/modules/depots/modals/depot.modal.js

```diff
@@ -0,0 +1,42 @@
+angular.module('bhima.controllers')
+  .controller('DepotModalController', DepotModalController);
+
+DepotModalController.$inject = [
+  '$state', 'ServiceService', 'DepotService',
+  'SessionService', 'ModalService', 'NotifyService',
+];
+
+function DepotModalController($state, Services, Depots,
+  SessionService, ModalService, Notify) {
+  var vm = this;
+
+  vm.depot = $state.params.depot;
+  vm.isCreating = !!($state.params.creating);
+
+  // exposed methods
+  vm.submit = submit;
+  vm.closeModal = closeModal;
+
+  // submit the data to the server from all two forms (update, create)
+  function submit(depotForm) {
+    var promise;
+
+    if (depotForm.$invalid || !depotForm.$dirty) { return 0; }
```

Comment:
FYI, an alias for `!FormController.$dirty` is `FormController.$pristine`.  It makes the code easier to read.